src/components/StakingPosition/StakingPosition.tsx

```diff
@@ -0,0 +1,107 @@
+import { BigNumber, ethers } from "ethers"
+import React from "react"
+import useSherlock from "../../hooks/useSherlock"
+import useWaitTx from "../../hooks/useWaitTx"
+import { PERIODS_IN_SECONDS } from "../../pages/Staking"
+import { convertSecondsToDurationString } from "../../utils/time"
+import { Button } from "../Button/Button"
+import styles from "./StakingPosition.module.scss"
+
+interface Props {
+  /**
+   * Position ID
+   */
+  id: number
+}
+
+const StakingPosition: React.FC<Props> = ({ id }) => {
+  const [lockedForSeconds, setLockedForSeconds] = React.useState<number>()
+  const [usdcBalance, setUsdcBalance] = React.useState<BigNumber>()
+  const [sherRewards, setSherRewards] = React.useState<BigNumber>()
+  const [stakingPeriod, setStakingPeriod] = React.useState<number>()
+  const isUnlocked = lockedForSeconds && lockedForSeconds <= 0
+
+  const { getPositionLockupEnd, getPositionSherRewards, getPositionUsdcBalance, unstake, restake } = useSherlock()
+  const { waitForTx } = useWaitTx()
+
+  /**
+   * Fetch the timestamp at which the position
+   * can be unstaked or restaked.
+   */
+  const handleFetchLockupPeriod = React.useCallback(async () => {
+    const lockupEnds = await getPositionLockupEnd(id)
+
+    const timeLeft = lockupEnds.sub(BigNumber.from((new Date().valueOf() / 1000).toFixed(0))).toNumber()
+    setLockedForSeconds(timeLeft)
+  }, [getPositionLockupEnd, id])
+
+  /**
+   * Fetch position's current USDC balance, claimable at the end
+   * of the lockup period.
+   */
+  const handleFetchUsdcBalance = React.useCallback(async () => {
+    const positionUsdcBalance = await getPositionUsdcBalance(id)
+    setUsdcBalance(positionUsdcBalance)
+  }, [getPositionUsdcBalance, id])
+
+  /**
+   * Fetch position's guaranteed SHER rewards
+   */
+  const handleFetchSherRewards = React.useCallback(async () => {
+    const positionSherRewards = await getPositionSherRewards(id)
+    setSherRewards(positionSherRewards)
+  }, [getPositionSherRewards, id])
+
+  /**
+   * Unstake position
+   */
+  const handleUnstake = React.useCallback(async () => {
+    await waitForTx(async () => await unstake(id))
+  }, [unstake, id, waitForTx])
+
+  /**
+   * Restake position
+   */
+  const handleRestake = React.useCallback(async () => {
+    if (!stakingPeriod) {
+      return
+    }
+
+    await waitForTx(async () => await restake(id, stakingPeriod))
+  }, [restake, id, stakingPeriod, waitForTx])
+
+  React.useEffect(() => {
+    handleFetchLockupPeriod()
+    handleFetchSherRewards()
+    handleFetchUsdcBalance()
```

Comment:
nit: I typically use the prefix `handle` when it comes to event handlers. Not really the end of the world of course, but wondering if we should stick to that convention?

Reply:
You are right! I've been using the same convention. I think from writing a lot of event handlers I have a tendency to prefix all functions with `handle` haha. I'll update them.